src/main/scala/com/softwaremill/realworld/articles/ArticlesRepository.scala

```diff
@@ -123,17 +116,23 @@ class ArticlesRepository(quill: SqliteZioJdbcContext[SnakeCase], dataSource: Dat
   ).unit
     .provide(dsLayer)
 
-  def updateBySlug(updateData: ArticleData, slug: String): IO[Exception, Unit] = run(
-    queryArticle
-      .filter(_.slug == lift(slug.toLowerCase()))
-      .update(
-        record => record.slug -> lift(updateData.slug),
-        record => record.title -> lift(updateData.title),
-        record => record.description -> lift(updateData.description),
-        record => record.body -> lift(updateData.body)
-      )
-  ).unit
-    .mapError(_ => Exceptions.AlreadyInUse("Article name already exists"))
+  def updateBySlug(updateData: ArticleData, slug: String): IO[Exception, Unit] = (for {
+    _ <- run(
```

Comment:
As per what we discussed this code will be problematic after we enable FOREIGN KEY enforcement. This is because `slug` column can be altered however it's a `primary key` for article table. This will need to change in order to avoid pain. Also if we try to introduce artificial ID column then there would be no need to update `slug` text in other places but just in one.

Reply:
this is valid after recent PR was merged (with activating foreign keys logic in db)
fixed :)